src/Sylius/Bundle/RbacBundle/Form/EventSubscriber/AddParentFormSubscriber.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\RbacBundle\Form\EventSubscriber;
+
+use Sylius\Bundle\RbacBundle\Form\Exception\UnexpectedTypeException;
+use Sylius\Component\Rbac\Model\PermissionInterface;
+use Sylius\Component\Rbac\Model\RoleInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class AddParentFormSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @param string $type
+     */
+    public function __construct($type)
+    {
+        $this->type  = $type;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            FormEvents::PRE_SET_DATA => 'preSetData',
+        );
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSetData(FormEvent $event)
+    {
+        $resource = $event->getData();
+        $form = $event->getForm();
```

Comment:
Form is not used if resource is empty or is not instance of selected model, so this should be moved after those checks.